src/popupmenu.c

```diff
@@ -416,6 +416,114 @@ pum_under_menu(int row, int col, int only_redrawing)
 	    && col < pum_col + pum_width + pum_scrollbar;
 }
 
+/*
+ * displays text on the popup menu with specific attributes.
+ */
+    static void
+pum_screen_put_with_attr(int row, int col, char_u *text, int textlen, int attr)
+{
+    int		i;
+    int		leader_len;
+    int		char_len;
+    int		cells;
+    int		new_attr;
+    char_u	*rt_leader = NULL;
+    char_u	*match_leader = NULL;
+    char_u	*ptr = text;
+    garray_T	*ga = NULL;
+    char_u	*leader = ins_compl_leader();
+    unsigned	cur_cot_flags = get_cot_flags();
+    int		in_fuzzy = (cur_cot_flags & COT_FUZZY) != 0;
+
+#ifdef FEAT_RIGHTLEFT
+    if (leader != NULL && curwin->w_p_rl)
+        rt_leader = reverse_text(leader);
+#endif
+    match_leader = rt_leader != NULL ? rt_leader : leader;
+    leader_len = match_leader ? (int)STRLEN(match_leader) : 0;
+
+    if (match_leader != NULL && leader_len > 0)
+    {
+        if (!in_fuzzy)
+        {
+            ga = ALLOC_ONE(garray_T);
+            ga_init2(ga, sizeof(int[2]), 10);
+            char_u *match = mb_strstr(text, match_leader);
+            while (match)
+            {
+                if (ga_grow(ga, 1) == OK)
+                {
+                    int *p = ((int *)ga->ga_data) + 2 * ga->ga_len;
+                    p[0] = (int)(match - text);     // start position
+                    p[1] = p[0] + leader_len;       // end position
+                    ga->ga_len++;
+                }
+                match = mb_strstr(match + leader_len, match_leader);
+            }
+        }
+        else
+        {
+#ifdef FEAT_SEARCH_EXTRA
```

Comment:
I don't think we need the `#ifdef FEAT_SEARCH_EXTRA` here, since you added those ifdefs in the function as well. So we can ge trid of those here.